src/skeptic/lib.rs

```diff
@@ -166,22 +166,35 @@ enum Buffer {
 }
 
 fn extract_tests_from_file(path: &Path) -> Result<DocTest, IoError> {
-    let mut tests = Vec::new();
-    // Oh this isn't actually a test but a legacy template
-    let mut old_template = None;
-
     let mut file = try!(File::open(path));
     let ref mut s = String::new();
     try!(file.read_to_string(s));
-    let mut parser = Parser::new(s);
-
-    let mut buffer = Buffer::None;
 
     let ref file_stem = sanitize_test_name(path.file_stem().unwrap().to_str().unwrap());
-    let mut section = None;
+    // Oh this isn't actually a test but a legacy template
+    let mut old_template = None;
 
     // In order to call get_offset() on the parser,
     // this loop must not hold an exclusive reference to the parser.
```

Comment:
this comment is no longer valid here. It should be either moved along with the mentioned loop ;) or removed if it is no longer valid.